engine/resource/src/resource.h

```diff
@@ -406,6 +408,16 @@ namespace dmResource
      */
     const char* ResultToString(Result result);
 
+    struct PreloadHintInfo
+    {
+        HPreloader m_Preloader;
+        int32_t m_Parent;
+    };
+
+    // load with default internal buffer and its management, returns buffer ptr in 'buffer'
+    Result LoadResource(HFactory factory, const char* path, const char* original_name, void** buffer, uint32_t* resource_size);
+    // load with own buffer
+    Result DoLoadResource(HFactory factory, const char* path, const char* original_name, uint32_t* resource_size, LoadBufferType* buffer);
```

Comment:
The naming "Do..." suggests that it's a private function. (especially since one has to look at the code in the source file to figure out the difference)

Please rename it to e.g. LoadResourceFromBuffer